### PR TITLE
smsotp added

### PR DESCRIPTION
**Pull request by elilmatha**

### CODE REVIEW

File: smsotp/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/css/localstyles-ie7.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Pushpalanka) [inline smsotp/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/css/localstyles-ie7.css]: Update the license.. Please check rest of the files too.. This should be 2015

File: smsotp/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/SMSOTPAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    /**
+     * Check whether the authentication or logout request can be handled by the authenticator
+     */
+//    @Override
+    public boolean canHandle(HttpServletRequest request) {
+        if (log.isDebugEnabled()) {
+            log.debug("Inside SMSOTPAuthenticator canHandle method");
+        }
+        return !StringUtils.isEmpty(request.getParameter(SMSOTPConstants.CODE));
```
Comment (Pushpalanka) [inline smsotp/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/SMSOTPAuthenticator.java]: Use 'StringUtils.isNotEmpty()'

File: smsotp/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/SMSOTPAuthenticator.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+            context.setSubject(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier("an authorised user"));
+        } else {
+            throw new AuthenticationFailedException("Code mismatch");
+        }
+    }
+
+    /**
+     * Get the friendly name of the Authenticator
+     */
+//    @Override
```
Comment (Pushpalanka) [inline smsotp/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/SMSOTPAuthenticator.java]: Follow Java doc comments format

**Comment by elilmatha** [comment]
closing this because its done to support IS-5.0.0

**Pull request closed without merging**